dowhy/gcm/distribution_change_robust.py

```diff
@@ -0,0 +1,759 @@
+
+        method = One of 'regression', 're-weighting', 'MR'. By default, 'MR'.
+
+        regressor = the regression estimator: a class supporting .fit and .predict methods.
+
+        classifier = the classification estimator: a class supporting .fit and .predict_probabilities methods.
+
+        calibrator = Optional, a method for probability calibration on a calibration set.
+                     This could be a regressor (e.g. sklearn.isotonic.IsotonicRegression) or
+                     a classifier (e.g. sklearn.LogisticRegression).
+                     No need to do this if classifier is a sklearn.calibration.CalibratedClassifierCV learner.
+
+        X_calib = (n_calib, K) np.array with the X data (explanatory variables) for the calibration set.
+        T_calib = (n_calib,) np.array with the T data (sample indicator) for the calibration set.
+        w_train = optional (n_calib,) np.array with sample weights for the calibration set.
+
+        all_indep = boolean, True if all explanatory variables are independent (used for self._simplify_C).
+        crop = float, all predicted probabilities from the classifier will be cropped below at this lower bound,
+               and above at 1-crop.
+
+        Returns:
+        theta_hat = the point estimate, np.mean(theta_scores) for the scores computed by self.est_scores.
+        std_err = the standard error for theta_hat, sem(theta_scores) for the scores computed by self.est_scores.
+        """
+
+        theta_scores = self.est_scores(
+            X_eval,
+            y_eval,
+            T_eval,
+            X_train,
+            y_train,
+            T_train,
+            w_eval=w_eval,
+            w_train=w_train,
+            method=method,  # One of 'regression', 're-weighting', 'MR',
+            regressor=regressor,
+            classifier=classifier,
+            calibrator=calibrator,
+            X_calib=X_calib,
+            T_calib=T_calib,
+            w_calib=w_calib,
+            all_indep=all_indep,
+            crop=crop,
+        )
+
+        if w_eval is not None:
+            w_sort = np.concatenate((w_eval[T_eval == 0], w_eval[T_eval == 1]))  # Order weights in same way as scores
+        else:
+            w_sort = np.ones(np.shape(T_eval))
+        weighted_stats = DescrStatsW(theta_scores, weights=w_sort, ddof=0)
+
+        return weighted_stats.mean, weighted_stats.std_mean
+
+
+def distribution_change_robust(
+    causal_model: ProbabilisticCausalModel,
+    old_data: pd.DataFrame,
+    new_data: pd.DataFrame,
+    target_node: Any,
+    sample_weight=None,
+    xfit=True,
+    xfit_folds=5,
+    train_size=0.5,
+    calib_size=0.0,
+    split_random_state=0,
+    method="MR",  # One of 'regression', 're-weighting', 'MR',
+    regressor: PredictionModel = create_linear_regressor,
+    classifier: ClassificationModel = create_logistic_regression_classifier,
+    calibrator: Optional[PredictionModel] = None,
+    variance=False,
```

Comment:
Maybe rather introduce a string parameter here (i.e. type of attribution). So, for now this would only take `"mean"` or `"variance"`. In the future, we can add something like `"wasserstein"` as well.

Reply:
Great idea! Done.